frontend/views/pages/GroupDashboard.vue

```
@@ -126,6 +130,11 @@ export default ({
     },
     updateBannerVisibility () {
       this.ephemeral.hideBanner = localStorage.getItem(this.bannerStorageKey) === 'true'
+    },
+    async requestPermissions () {
+      await requestNotificationPermission()
```

If this call ever throwed for some reason (for example in development when modifying the associated code) the `mousemove` listener would not get removed, likely causing a loop.
This can be avoided with a try/finally block.

---

Nice catch. Thanks. @snowteamer

---

What do you think if we do `removeEventListener` in `beforeDestroy` lifecycle?

---

But `requestNotificationPermission` function can not throw any errors.
I think try/catch/finally isn't needed here.

---

@Silver-IT But @snowteamer raises a good point - at some point in the future someone else could rewrite your code for `requestNotificationPermission` so that it does throw an exception. In that case this part of the app will break in a serious way.

---

Since the current `requestNotificationPermission` function can not throw any errors, should we leave this without any changes? I think I should add comment here but to be honest, I don't think I should use try/finally at the moment.

---

It is common/encouraged for all `await` statements to be wrapped in a `try`/`catch` somewhere above where they're called, so that uncaught exceptions do not occur. Again, if this is not wrapped in such a handler, then we open ourselves to forgetting that this exists at all, and a future change to `requestNotificationPermission` can potentially break the app in a serious way.

We should catch all exceptions where they might occur, even if right now you're confident that no exceptions shouldn't happen. It's part of our policy to catch all possible errors and handle them appropriately.

---

And in this case, if there's an exception, I believe @snowteamer's suggestion is to remove the event listener so that it doesn't get stuck in a loop. That sounds like a reasonable/wise course of action to me.

---

Understand. Thanks, @taoeffect , @snowteamer.